addons/base_vat/models/res_partner.py

```diff
@@ -224,6 +224,27 @@ def check_vat_al(self, vat):
             return True
         return False
 
+    __check_vat_hu_companies_re = re.compile(r'^8\d{9}$')
+    __check_vat_hu_individual_re = re.compile(r'^\d{8}-[12345]-[0-9]{2}$')
+
+    def check_vat_hu(self, vat):
+        """
+            Check Hungary VAT number that can be for example 'HU12345676 or 'xxxxxxxx-y-zz' or '8xxxxxxxxy'
+            - For xxxxxxxx-y-zz, 'x' can be any number, 'y' is a number between 1 and 5 depending on the person and the 'zz'
+              is used for region code.
+            - 8xxxxxxxxy, Tin number for individual, it has to start with an 8 and finish with the check digit
+        """
+        companies = self.__check_vat_hu_companies_re.match(vat)
+        if companies:
+            return True
+        individual = self.__check_vat_hu_individual_re.match(vat)
+        if individual:
+            return True
+        elif len(vat) == 10 and vat[:2] == 'HU':
+            return True
```

Comment:
`stdnum` seems to handle the checksum for regular numbers
https://github.com/arthurdejong/python-stdnum/blob/master/stdnum/hu/anum.py

We would be losing this feature if we only check for the length. It also allows inputting only the digits without the country code prefix.